controllers/blogs.js

```diff
@@ -1,39 +1,46 @@
 const blogsRouter = require("express").Router();
 const Blog = require("../models/blog");
-
 const logger = require("../utils/logger");
 
-blogsRouter.get("/", (request, response) => {
-  Blog.find({})
-    .then(blogs => {
-      return (response.json(blogs));
-    });
+blogsRouter.get("/", async (request, response) => {
+  try {
+    const blogs = await Blog.find({});
+    response.json(blogs);
+  } catch (error) {
+    response.status(500).json({ error: "Something went wrong" });
+  }
 });
 
-blogsRouter.get("/:id", (request, response) => {
-  Blog.findById(request.params.id)
-    .then(foundBlog => {
-      return (response.json(foundBlog));
-    });
+blogsRouter.get("/:id", async (request, response) => {
+  try {
+    const foundBlog = await Blog.findById(request.params.id);
+    response.json(foundBlog);
+  } catch (error) {
+    response.status(404).json({ error: "Blog not found" });
+  }
 });
 
-blogsRouter.delete("/:id", (request, response) => {
-  Blog.findByIdAndRemove(request.params.id)
-    .then(deletedBlog => {
-      return (response.status(200).json(deletedBlog));
-    });
+blogsRouter.delete("/:id", async (request, response) => {
+  try {
+    const deletedBlog = await Blog.findByIdAndRemove(request.params.id);
+    response.status(200).json(deletedBlog);
+  } catch (error) {
+    response.status(404).json({ error: "Blog not found" });
+  }
 });
 
-blogsRouter.post("/", (request, response) => {
-  logger.info(request.json);
-  const blog = new Blog(request.body);
-  logger.info(blog);
-  logger.info(request.body);
+blogsRouter.post("/", async (request, response) => {
+  try {
+    logger.info(request.json);
+    const blog = new Blog(request.body);
+    logger.info(blog);
+    logger.info(request.body);
 
-  blog.save()
-    .then(result => {
-      response.status(201).json(result);
-    });
+    const result = await blog.save();
+    response.status(201).json(result);
+  } catch (error) {
+    response.status(400).json({ error: "Bad request" });
```

Comment:
```suggestion
    next(error);
```
And here it is just calling next(error) which will be picked up by the errorHandler.
I think if you just merge main to this branch and then solve the conflicts it should work.